File: packages/rhiz-sdk-py/rhiz_sdk/api/analytics.py
"""
Analytics API operations
"""

from typing import Any

import httpx


class AnalyticsAPI:
    """Analytics API client"""

    def __init__(self, client: httpx.Client):
        self._client = client

    def get_trust_health(self, entity_id: str) -> dict[str, Any]:
        """
        Get trust health metrics for an entity

        Args:
            entity_id: Entity ID

        Returns:
            Trust health response
        """
        response = self._client.get(f"/analytics/trust-health/{entity_id}")
        response.raise_for_status()
        return response.json()

    def get_trust_metrics(self, entity_id: str) -> dict[str, Any]:
        """
        Get detailed trust metrics for an entity

        Args:
            entity_id: Entity ID

        Returns:
            Trust metrics response
        """
        response = self._client.get(f"/analytics/trust-metrics/{entity_id}")
        response.raise_for_status()
        return response.json()

    def get_network_stats(self) -> dict[str, Any]:
        """
        Get overall network statistics

        Returns:
            Network stats response
        """
        response = self._client.get("/analytics/network-stats")
        response.raise_for_status()
        return response.json()

File: rhiz/packages/sdk-ts/src/api/analytics.ts
/**
 * Analytics API operations
 */

import type { AxiosInstance } from 'axios';
import type { TrustMetrics } from '@rhiz/protocol';

export interface TrustHealthResponse {
  entity_id: string;
  trust_level: string;
  trust_score: number;
  network_size: number;
  verified_ratio: number;
  recent_activity: number;
  recommendations: string[];
}

export interface NetworkStatsResponse {
  total_entities: number;
  total_relationships: number;
  avg_trust_score: number;
  verified_entities: number;
}

export class AnalyticsAPI {
  constructor(private client: AxiosInstance) {}

  async getTrustHealth(entityId: string): Promise<TrustHealthResponse> {
    const response = await this.client.get(`/analytics/trust-health/${entityId}`);
    return response.data;
  }

  async getTrustMetrics(entityId: string): Promise<TrustMetrics> {
    const response = await this.client.get(`/analytics/trust-metrics/${entityId}`);
    return response.data;
  }

  async getNetworkStats(): Promise<NetworkStatsResponse> {
    const response = await this.client.get('/analytics/network-stats');
    return response.data;
  }
}

